app/main.py

```diff
@@ -2,0 +1,48 @@
+class SoftwareEngineer:
+    def __init__(self, name: str) -> None:
+        self.skills = []
+        self.name = name
+
+    def learn_skill(self, skill: str) -> None:
+        self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.skills = ["JavaScript", "HTML", "CSS"]
+
+    def create_awesome_web_page(self) -> str:
+        print(f"{self.name} is creating a webpage...")
+        return "<h1>Hello world</h1>"
+
+
+class BackendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.skills = ["Python", "SQL", "Django"]
```

Comment:
Instead of reassigning `self.skills`, consider using `self.skills.extend(['Python', 'SQL', 'Django'])` to add skills to the existing list. This prevents overwriting the skills list initialized in the `SoftwareEngineer` class.